src/evaluate/evaluator/text_classification.py

```diff
@@ -31,77 +58,100 @@ class TextClassificationEvaluator(Evaluator):
     def __init__(self, task="text-classification", default_metric_name=None):
         super().__init__(task, default_metric_name=default_metric_name)
 
+    def prepare_data(self, data: Union[str, Dataset], input_column: str, input_column2: str, label_column: str):
+        if data is None:
+            raise ValueError(
+                "Please specify a valid `data` object - either a `str` with a name or a `Dataset` object."
+            )
+        data = load_dataset(data) if isinstance(data, str) else data
+        if input_column not in data.column_names:
+            raise ValueError(
+                f"Invalid `input_column` {input_column} specified. The dataset contains the following columns: {data.column_names}."
+            )
+        if input_column2 is not None and input_column2 not in data.column_names:
+            raise ValueError(
+                f"Invalid `input_column2` {input_column2} specified. The dataset contains the following columns: {data.column_names}."
+            )
+        if label_column not in data.column_names:
+            raise ValueError(
+                f"Invalid `label_column` {label_column} specified. The dataset contains the following columns: {data.column_names}."
+            )
+
+        return {"references": data[label_column]}, DatasetColumnTextClassification(data, input_column, input_column2)
+
     def predictions_processor(self, predictions, label_mapping):
         predictions = [
             label_mapping[element["label"]] if label_mapping is not None else element["label"]
             for element in predictions
         ]
         return {"predictions": predictions}
 
-    def compute(self, *args, **kwargs) -> Tuple[Dict[str, float], Any]:
+    @add_start_docstrings(EVALUTOR_COMPUTE_START_DOCSTRING)
+    @add_end_docstrings(EVALUATOR_COMPUTE_RETURN_DOCSTRING, TASK_DOCUMENTATION)
+    def compute(
+        self,
+        model_or_pipeline: Union[
+            str, "Pipeline", Callable, "PreTrainedModel", "TFPreTrainedModel"  # noqa: F821
+        ] = None,
+        data: Union[str, Dataset] = None,
+        metric: Union[str, EvaluationModule] = None,
+        tokenizer: Optional[Union[str, "PreTrainedTokenizer"]] = None,  # noqa: F821
+        feature_extractor: Optional[Union[str, "FeatureExtractionMixin"]] = None,  # noqa: F821
+        strategy: Literal["simple", "bootstrap"] = "simple",
+        confidence_level: float = 0.95,
+        n_resamples: int = 9999,
+        device: int = None,
+        random_state: Optional[int] = None,
+        input_column: str = "text",
+        input_column2: Optional[str] = None,
+        label_column: str = "label",
+        label_mapping: Optional[Dict[str, Number]] = None,
+    ) -> Tuple[Dict[str, float], Any]:
+        """
+        input_column (`str`, *optional*, defaults to `"text"`):
+            the name of the column containing the text feature in the dataset specified by `data`.
+        input_column2 (`str`, *optional*, defaults to `None`):
```

Comment:
i would avoid numbers in variable names, especially in signatures.

```suggestion
        second_input_column (`str`, *optional*, defaults to `None`):
```